### PR TITLE
Prepare for 0.15.0 release

### PR DESCRIPTION
**Pull request by torkleyy**
## Checklist

* [x] I've added tests for all code changes and additions (where applicable)
* [x] I've added a demonstration of the new feature to one or more examples
* [x] I've updated the book to reflect my changes
* [x] Usage of new public items is shown in the API docs

## API changes

* Remove multiple methods from `WorldExt` which were moved to `shred::World` (see https://github.com/slide-rs/shred/pull/141)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,19 +24,19 @@ travis-ci = { repository = "slide-rs/specs" }
...
-shred = { version = "0.8.0", default-features = false }
+shred = { version = "0.9.0", default-features = false }
 shrev = "1.0.0"
-shred-derive = "0.6.0"
 tuple_utils = "0.3"
-rayon = { version = "1.0.0", optional = true }
 nonzero_signed = "1.0.1"
-uuid = { version = "0.7.4", optional = true, features = ["v4", "serde"] }
 
+rayon = { version = "1.0.0", optional = true }
```
Comment (azriel91) [inline Cargo.toml]: Shall we use `rayon = "1.1.0"`, which uses `crossbeam-deque 0.6.3`?

File: src/world/world_ext.rs [hunk 2]
```diff
@@ -119,116 +140,11 @@ pub trait WorldExt {
...
-    ///
-    /// * `LazyUpdate` and
-    /// * `EntitiesRes`
-    ///
-    /// Both of them should only be fetched immutably, which is why
-    /// the latter one has a type def for convenience: `Entities` which
-    /// is just `Fetch<EntitiesRes>`. Both resources are special and need
-    /// to execute code at the end of the frame, which is done in
-    /// `World::maintain`.
+    /// **DEPREACTED:** Use [World::insert] instead.
```
Comment (azriel91) [inline src/world/world_ext.rs]: typo: DEPRECATED

Comment (torkleyy) [reply src/world/world_ext.rs]: Thanks, fixed.

**Comment by torkleyy** [comment]
bors delegate=azriel91

**Comment by azriel91** [comment]
FYI I'm making changes locally.

**Comment by azriel91** [comment]
Pending https://github.com/torkleyy/specs/pull/1

**Pull request merged**